python/cudf/cudf/core/tools/numeric.py

```diff
@@ -0,0 +1,241 @@
+        hasattr(arg, "ndim") and arg.ndim > 1
+    ):
+        raise ValueError("arg must be column convertible")
+
+    col = as_column(arg)
+    dtype = col.dtype
+
+    if is_datetime_dtype(dtype) or is_timedelta_dtype(dtype):
+        col = col.as_numerical_column(np.dtype("int64"))
+    elif is_categorical_dtype(dtype):
+        cat_dtype = col.dtype.type
+        if is_numerical_dtype(cat_dtype):
+            col = col.as_numerical_column(cat_dtype)
+        else:
+            try:
+                col = _convert_str_col(col._get_decategorized_column(), errors)
+            except ValueError as e:
+                if errors == "ignore":
+                    return arg
+                else:
+                    raise e
+    elif is_string_dtype(dtype):
+        try:
+            col = _convert_str_col(col, errors)
+        except ValueError as e:
+            if errors == "ignore":
+                return arg
+            else:
+                raise e
+    elif is_list_dtype(dtype) or is_struct_dtype(dtype):
+        raise ValueError("Input does not support nested datatypes")
+    elif is_numerical_dtype(dtype):
+        pass
+    else:
+        raise ValueError("Unrecognized datatype")
+
+    if downcast:
+        downcast_type_map = {
+            "integer": list(np.typecodes["Integer"]),
+            "signed": list(np.typecodes["Integer"]),
+            "unsigned": list(np.typecodes["UnsignedInteger"]),
+        }
+        float_types = list(np.typecodes["Float"])
+        idx = float_types.index(np.dtype(np.float32).char)
+        downcast_type_map["float"] = float_types[idx:]
+
+        type_set = downcast_type_map[downcast]
+        is_similar_kind_cast = (
+            col.dtype.kind == "f" and downcast == "float"
+        ) or (
+            col.dtype.kind in {"i", "u"}
+            and downcast in {"integer", "signed", "unsigned"}
+        )
+
+        for t in type_set:
+            downcast_dtype = np.dtype(t)
+            if downcast_dtype.itemsize <= col.dtype.itemsize:
+                if is_similar_kind_cast:
+                    if col.can_cast_safely(downcast_dtype):
+                        col = libcudf.unary.cast(col, downcast_dtype)
+                        break
+                else:
+                    if _can_cast_no_overflow(col, downcast_dtype):
+                        # Round off for float -> (u)int
+                        if col.dtype.kind == "f" and downcast_dtype.kind in {
+                            "i",
+                            "u",
+                        }:
+                            # TODO: use libcudf round once
+                            # https://github.com/rapidsai/cudf/pull/6562 merges
+                            col = col.round()
+                        col = libcudf.unary.cast(col, downcast_dtype)
+                        break
+
+    if isinstance(arg, (cudf.Series, pd.Series)):
+        return cudf.Series(col)
+    else:
+        return col.to_array(fillna="pandas")
```

Comment:
Can we use the explicit `fillna`? I think it's an open TODO to remove the `to_array` method and just use `.values` or `cupy.asarray`